modules/Bio/EnsEMBL/Variation/Pipeline/AncestralAlleles/Assign.pm

```diff
@@ -0,0 +1,168 @@
+=cut
+
+package Bio::EnsEMBL::Variation::Pipeline::AncestralAlleles::Assign;
+
+use strict;
+use warnings;
+
+use FileHandle;
+use Bio::EnsEMBL::Registry;
+use File::Path qw(make_path);
+use Data::Dumper;
+use Bio::DB::Fasta;
+use base qw(Bio::EnsEMBL::Variation::Pipeline::BaseVariationProcess);
+
+sub run {
+  my $self = shift;
+  my $species_name = $self->param('species_name');
+  my $batch = $self->param('batch');
+  my $id = $batch->{id};
+  my $start = $batch->{start};
+  my $end = $batch->{end};
+  my $ancestral_file = $self->param('ancestral_file');
+
+  $self->setup();
+  $self->dump_variation_features();
+  $self->assign_ancestral_alleles();
+  $self->cleanup();
+}
+
+sub setup {
+  my $self = shift;
+  my $batch = $self->param('batch');
+  my $id = $batch->{id};
+  my $compara_dir = $self->param('compara_dir'); 
+  my $species_dir = $self->param('species_dir');
+  my $ancestral_file_targz = $self->param('ancestral_file');
+  my $ancestral_file_notargz = $ancestral_file_targz;
+  $ancestral_file_notargz =~ s/\.tar\.gz//;
+  # create temp directory for storing and uncompressing ancestral fasta file 
+  unless (-d "$species_dir/fasta_$id") {
+    my $err;
+    make_path("$species_dir/fasta_$id", {error => \$err});
+    die "make_path failed: ".Dumper($err) if $err && @$err;
+  }
+  # cp  ancestral fasta file from compara dir to temp directory
+  $self->run_cmd("cp $compara_dir/$ancestral_file_targz $species_dir/fasta_$id");
+  # uncompress ancestral fasta file
+  $self->run_cmd("tar xzf $species_dir/fasta_$id/$ancestral_file_targz -C $species_dir/fasta_$id/");
+  # rm bed files
+  $self->run_cmd("rm $species_dir/fasta_$id/$ancestral_file_notargz/*.bed");
+  my $fasta_files_dir = "$species_dir/fasta_$id/$ancestral_file_notargz"; 
+  $self->param('fasta_files_dir', $fasta_files_dir);
+  my $fasta_files_dir_root = "$species_dir/fasta_$id/"; 
+  $self->param('fasta_files_dir_root', $fasta_files_dir_root);
+}
+
+sub dump_variation_features {
+  my $self = shift;
+  my $species_name = $self->param('species_name');
+  my $species_dir = $self->param('species_dir');
+  my $batch = $self->param('batch'); 
+  my ($start, $end) = ($batch->{start}, $batch->{end});
+  my $vf_file = "$species_dir/vf_$start\_$end";
+  my $aa_file = "$species_dir/aa_$start\_$end";
+  $self->param('vf_file', $vf_file);
+  $self->param('aa_file', $aa_file);
+  if (! -e "$vf_file") {
+    my $fh = FileHandle->new($vf_file, 'w');
+    $self->param('species', $species_name);
+    my $vdba = $self->get_species_adaptor('variation');
+    my $dbc = $vdba->dbc;
+    my $sth = $dbc->prepare(qq{
+      SELECT vf.variation_feature_id, sr.name, vf.seq_region_start, vf.seq_region_end, vf.seq_region_strand, vf.ancestral_allele
+      FROM variation_feature vf, seq_region sr
+      WHERE vf.seq_region_id = sr.seq_region_id
+      AND vf.variation_feature_id >= ?
+      AND vf.variation_feature_id <= ?
+      ORDER BY vf.variation_feature_id;}, {mysql_use_result => 1}
```

Comment:
Is the order by quick enough for how useful it is?

Reply:
This is not needed anymore now that we annotate per variation feature. It was relevant for the old script. I will remove that. But it is not a bottleneck.